### DISCUSSION TITLE
Add delay before each request

### DISCUSSION THREAD

**Discussion by primeapple** [mswjs/msw#2060] [Q&A] [answered]
### Scope

Adds a new behavior

### Compatibility

- [ ] This is a breaking change

### Feature description

I want to have what was described here https://github.com/mswjs/msw/issues/260 , but isn't usable anymore since V2.

Basically add a `delay()` before each request. What would the suggested approach to this be?

I tried using the lifecycle methods but had no success.

**Comment by ZeroCho** [comment]
https://mswjs.io/docs/api/delay
Have you tried this one?

**Comment by primeapple** [comment]
Sure, but it requires me to manually set a delay for each handler, which is not what I want. I want a delay before EVERY request.

**Comment by mattcosta7** [comment]
You can do something like a higher-order resolver similar to these:

https://mswjs.io/docs/recipes/higher-order-resolver. It still requires manually wrapping each handler's resolver with that per handled request, but provides a good mechanism for handling those

**Comment by kettanaito** [comment, marked as answer]
Hi, @primeapple. 

Since this is a common question, I've added a new [Global response delay](https://mswjs.io/docs/recipes/global-response-delay) recipe to the docs. It will guide you through two main options on how to achieve the global delay you describe. Please refer to those options and choose the one that suits you the most.

**Reply by mattcosta7** [reply]
@kettanaito one nit here

https://mswjs.io/docs/recipes/global-response-delay#option-1-a-delay-passthrough-handler

`server.use` will apply handlers _before_ this so they'd need to have a separate delay defined

**Reply by kettanaito** [reply]
Good point. I think it borders with common knowledge of the library and how `server.use()` works. If folks have issues with this, I don't mind updating the page. I'd likely recommend storing the `http.all()` handler in a variable and providing it as the first handler if you want to use `server.use()` while also keep the global delay consistent.
